addons/account/models/account.py

```diff
@@ -178,6 +178,8 @@ def name_search(self, name, args=None, operator='ilike', limit=100):
     def onchange_internal_type(self):
         if self.internal_type in ('receivable', 'payable'):
             self.reconcile = True
+        else:
+            self.reconcile = False
```

Comment:
why not a simple:

self.reconcile = self.internal_type in ('receivable', 'payable')

Reply:
Ho yeah... It's a Friday afternoon commit... sorry!